core-bundle/src/EventListener/BackendMenuListener.php

```diff
@@ -12,82 +12,52 @@
 
 namespace Contao\CoreBundle\EventListener;
 
-use Contao\BackendUser;
+use Contao\Backend;
 use Contao\CoreBundle\Event\MenuEvent;
-use Knp\Menu\FactoryInterface;
-use Knp\Menu\ItemInterface;
+use Contao\CoreBundle\Framework\ContaoFramework;
 use Symfony\Component\Security\Core\Security;
 
-class BackendMenuListener
+final class BackendMenuListener
```

Comment:
Was that listener introduced in 4.9? If not, then this is theoretically a BC break, as is the additional argument in the constructor. I know we said listeners aren't really API but as long as it's technically possible, people will do it.

Reply:
well the listener is a complete rewrite and has very little to do with the previous one. But we never really discussed `final` and whether we should use it, so I don't think this should be in this PR.

Reply:
The same listener exists in the manager bundle and the class is final there (also in Contao 4.8). I guess I have copied it from there.

But @aschempp is right, we should not use `final` until we have discussed the concept. I will therefore adjust the two classes in the manager bundle as well.

Reply:
Fixed in 6f2250625f7325b1cf43119c356a5a7517102499.